### PR TITLE
Add interceptor support

### LINKED ISSUE #37: metadata keyword argument not respected by get_by_endpoint()
**Describe the bug**
metadata not respected by get_by_endpoint()

When trying to connect to a grpc server that requires a bearer token, I get a response as below...
`
grpc._channel._MultiThreadedRendezvous: <_MultiThreadedRendezvous of RPC that terminated with:
        status = StatusCode.PERMISSION_DENIED
        details = "RBAC: access denied"
        debug_error_string = "UNKNOWN:Error received from peer  {grpc_message:"RBAC: access denied", grpc_status:7, created_time:"2023-11-27T11:59:49.7471035+00:00"}"
`

**To Reproduce**
Following the example to use a bearer token in https://github.com/wesky93/grpc_requests/blob/master/src/examples/README.md
it looks like the `metadata` keyword argument is not consumed by any `__init__` method in the creation of the `Client` Object

**Expected behavior**
The bearer token is used and a connection is made to our server.

**Desktop (please complete the following information):**
Ubuntu 22.04 running a `python:3.12` derived docker container

**Additional context**
I'm fairly confident the issue is with the metadata keyword argument not being passed to the appropriate grpc calls under the hood.

Let me know if you need any more info,
Thanks

### PR DESCRIPTION
**Pull request by artificial-aidan**
Helps fix #37

### CODE REVIEW

File: requirements-dev.txt [hunk 1]
```diff
@@ -6,4 +6,4 @@ flake8>=5.0.4
 pytest-cov>=4.0.0
 pytest-asyncio>=0.15.1
 aiounittest==1.4.2
-
+grpc-interceptor==0.15.4
```
Comment (ViridianForge) [inline requirements-dev.txt:9]: Does the interceptor requirement need to be added to requirements.txt for it to be accessible for library users who may not have dev requirements added?

**Review by ViridianForge — COMMENTED** [review]
One question - and can you re-target the PR against the develop branch?

File: requirements-dev.txt (hunk 1 above)
Comment (artificial-aidan) [reply requirements-dev.txt:9]: The interceptor requirement is only for testing in this library. The interceptor suppose is built into the core grpc library, but the interceptor package has some extra data structures that help build interceptors. But it's not a requirement.

**Comment by imrimt** [comment]
I just ran into this using the `metadata` keyword. Looks like the docs still references this mechanism even though it doesn't seem to work. Can we get the updated example of how to use the interceptors for authorization like this?

**Pull request merged**